salt/pillar/__init__.py

```diff
@@ -456,7 +456,11 @@ def fetch_pillar(self):
             pillar_override=self.pillar_override,
             pillarenv=self.pillarenv,
         )
-        return fresh_pillar.compile_pillar()
+        pillars = fresh_pillar.compile_pillar()
+        # Could .pop instead, but LBYL is slightly faster
+        if "_errors" in pillars:
```

Comment:
Don't we still want the user to see the warning?:

```
local:
    ----------
    _errors:
        - Rendering SLS 'default' failed. Please see master log for details.
  ```

I understand not wanting the error in the actual pillar cache. I tested with your PR and I do not see the error anymore when I run pillar.items:

```
(heist-3.8)  ch3ll@megan-precision5550  ~/git/salt   fix/61010-pillar-cache-_errors  salt \* pillar.items           
/home/ch3ll/.pyenv/versions/3.8.6/envs/heist-3.8/lib/python3.8/site-packages/_distutils_hack/__init__.py:30: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
poc-minion:
    ----------
```

Reply:
Let me see if I can make this happen. I know that it does stick the error in the log... :thinking: